columnflow/production/cms/muon.py

```diff
@@ -14,6 +16,27 @@
 ak = maybe_import("awkward")
 
 
+@dataclass
+class MuonSFConfig:
+    correction: str
+    campaign: str
+
+    @classmethod
+    def new(
+        cls,
+        obj: MuonSFConfig | tuple[str, str],
+    ) -> MuonSFConfig:
+        # purely for backwards compatibility with the old tuple format
+        if isinstance(obj, cls):
+            return obj
+        elif isinstance(obj, list) or isinstance(obj, tuple) or isinstance(obj, set):
+            return cls(*obj)
+        elif isinstance(obj, dict):
+            return cls(**obj)
+        else:
+            raise ValueError(f"cannot convert {obj} to MuonSFConfig")
```

Comment:
De-nesting. Also, unordered sets are not supported.

```suggestion
        if isinstance(obj, cls):
            return obj
        if isinstance(obj, str):
            return cls(obj)
        if isinstance(obj, (list, tuple)):
            return cls(*obj)
        if isinstance(obj, dict):
            return cls(**obj)
        raise ValueError(f"cannot convert {obj} to MuonSFConfig")
```